alfresco-search/src/test/java/org/alfresco/solr/AbstractAlfrescoDistributedTest.java

```diff
@@ -346,15 +349,22 @@ public void waitForShardsCount(Query query, int count, long waitMillis, long sta
             totalCount = 0;
             for (SolrCore core : cores) {
                 RefCounted<SolrIndexSearcher> refCounted = null;
+                boolean refCountedDecremented = false;
                 try {
                     refCounted = core.getSearcher();
+                    
                     SolrIndexSearcher searcher = refCounted.get();
                     TopDocs topDocs = searcher.search(query, 10);
                     totalCount += topDocs.totalHits;
                     //System.out.println("####### shard count:"+core.getName()+":"+totalCount);
+                    refCounted.decref();
+                    refCountedDecremented = true;
                     Thread.sleep(2000);
                 } finally {
-                    refCounted.decref();
+                    if(refCounted!=null && !refCountedDecremented)
```

Comment:
Could we get rid of the boolean and always call `refCounted.decref()` here instead of in the try block?  Assuming `decref` can't throw an exception then this will be equivalent (but sometimes 2 seconds later).  If `decref` can throw an exception then I think we have other problems.

Reply:
I discussed this with @alessandrobenedetti but honestly I don't remember why he decided to follow this approach. I'll talk with him and let you know

Reply:
in the meantime I'll push a change for the formatting

Reply:
Added explanation

Reply:
Thank you!